cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -801,4 +805,27 @@ private Set<Class<?>> addExtraDependencies(CConfiguration cConf, Set<Class<?>> d
 
     return dependencies;
   }
+
+  @VisibleForTesting
+  static void copyRuntimeToken(LocationFactory locationFactory, Arguments args, File tempDir,
+                               Map<String, LocalizeResource> localizeResources) throws IOException {
+    if (args.hasOption(ProgramOptionConstants.PEER_NAME)) {
+      File runtimeTokenFile = File.createTempFile("runtime-tether", ".token", tempDir);
+      Location location = locationFactory.create(URI.create(args.getOption(ProgramOptionConstants
+                                                                             .PROGRAM_RESOURCE_URI)));
+      boolean runtimeTokenCopied = true;
+      try (InputStream is = location.append(Constants.Security.Authentication.RUNTIME_TOKEN_FILE).getInputStream()) {
+        Files.copy(is, runtimeTokenFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        LOG.debug("Copied {} to {}", Constants.Security.Authentication.RUNTIME_TOKEN_FILE, runtimeTokenFile);
+      } catch (IOException e) {
+        // File may not exist; in this case, skip copying the runtime token.
```

Comment:
when would this happen? Would it make more sense to fail here rather than continue?

Reply:
As this runtime token is passed via a remote instance which can have internal identity enforcement either enabled or disabled and there's technically no way for the HDF instance to know, the only way would be to check if the file was passed along with the initial program launch.

Reply:
I see. It would be clearer to check if it exists before trying to copy it then, instead of catching this exception. Should also not have a warning log if it's expected normal behavior.

Reply:
Understood, done. Thanks!